scripts/rewritepass.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python3
+import sys
+import os
+
+if len(sys.argv) != 3:
+  print("Use: %s <opt path> <passes>" % sys.argv[0])
+  exit(1)
+
+optpath = sys.argv[1]
+passes = sys.argv[2].strip("'\"").split(',')
+plen = len(passes)
+
+level_ancestors = {
+  "cgscc":["module"],
+  "function":["module"],
+      # don't use module(cgscc(function(...)), use module(function(...))
+      # the former one results in a different output.
+      # e.g. llvm/test/Transforms/LoopVectorize/reduction-order.ll
+  "loop":["function", "module"]
+}
+
+"""
+PassBuilder.cpp states that:
+  As a special shortcut, if the very first pass is not
+  a module pass (as a module pass manager is), this will automatically form
+  the shortest stack of pass managers that allow inserting that first pass.
+Build the stack of pass managers by invoking opt with the first pass and
+parsing its error message.
+"""
+i = 0
+firstp = passes[i]
```

Comment:
why do you need to split the pass list?
I thought it would be enough to get the pass string, run `opt -passes=$str,globalopt` and then apply the right conversion and then output smth like `function($str)`. We don't need to decorate every pass, right? LLVM supports the bundling.

Reply:
Because PassBuilder's comment is stating that if the very first pass is not a module pass it will do something.
I agree that intuitively the first pass's level should be equivalent to the whole sequence, but simply didn't want to rely on an additional reasoning :/
The comment also says that all the passes should be in the equivalent level, but I found that some passes are 'polymorphic' (ex: `-passes='verify,simplify-cfg,verify,globalopt'`), so I had a feeling that something unexpected might happen..

Reply:
Should we move to the direction for the simplicity of rewritepass.py? I'm not strongly with my opinion though.

Reply:
Maybe let's just pass the PassRegistry.def file and lookup the right decorators to use from there.
BTW, I don't think you need to add `module(foo)`. Those are NOPs AFAIU.